regression/check_style.sh

```diff
@@ -69,91 +86,194 @@ if [[ ! ${cf} ]]; then
+   fix_mode=0
+   ;;
 \?) echo "" ;echo "invalid option: -$OPTARG"; print_use; exit 1 ;;
 :)  echo "" ;echo "option -$OPTARG requires an argument."; print_use; exit 1 ;;
 esac
 done
 
 ##---------------------------------------------------------------------------##
-## Check mode
+## Check mode (Test C++ code with git-clang-format)
 ##---------------------------------------------------------------------------##
 
-if test "${pct_mode}" = "1"; then
+ver=`${cf} --version`
+echo -e "\n--------------------------------------------------------------------------------"
+echo "Checking modified C/C++ code for style conformance..."
+#echo "  - using clang-format version $ver"
+#echo "  - using settings from this project's .clang_format configuration file."
+echo " "
+
+patchfile_c=$(mktemp /tmp/gcf.patch.XXXXXXXX)
 
-  # don't actually modify the files (compare to branch 'develop')
-  cmd='${gcf} --binary ${cf} -f --diff --extensions hh,cc develop'
-  echo "Running..."
-  echo "   ${gcf} --binary ${cf} -f --diff --extensions hh,cc develop"
-  echo " "
-  result=`eval $cmd`
-  allok=`echo $result | grep -c "did not modify"`
-  # 2nd chance (maybe there are no files to check)
-  if test $allok = 0; then
-    allok=`echo $result | grep -c "no modified files"`
-  fi
+# don't actually modify the files (compare to branch 'develop')
+cmd="${gcf} --binary ${cf} -f --diff --extensions hh,cc develop"
+run "${cmd}" &> $patchfile_c
 
-  if test $allok = 1; then
-    echo "PASS: Changes conform to this project's style requirements."
+# if the patch file has the string "no modified files to format", the check passes.
+if [[ `grep -c "no modified files" ${patchfile_c}` != 0 ]]; then
+  echo "PASS: Changes to C++ sources conform to this project's style requirements."
+else
+  foundissues=1
+  echo    "FAIL: some C++ files do not conform to this project's style requirements:"
+  # Modify files, if requested
+  if [[ ${fix_mode} == 1 ]]; then
+    echo -e "      The following patch has been applied to your file.\n"
+    run "git apply $patchfile_c"
+    cat $patchfile_c
   else
-    echo "FAIL: some files do not conform to this project's style requirements:"
-    echo " "
-    # rerun the command to capture color output.
-    eval $cmd
-    exit 1
+    echo -e "      run ${0##*/} with option -f to automatically apply this patch.\n"
+    cat $patchfile_c
   fi
+fi
+rm -f $patchfile_c
 
 ##---------------------------------------------------------------------------##
-## Fix mode
-##   no options --> fix the files by running clang-format
-##   -d | -n    --> print diff of required changes.
+## Check mode (Test F90 code indentation with emacs and bash)
 ##---------------------------------------------------------------------------##
 
-else
+# Defaults ----------------------------------------
+EMACS=`which emacs`
+if [[ $EMACS ]]; then
+  EMACSVER=`$EMACS --version | head -n 1 | sed -e 's/.*Emacs //'`
+  if [[ `version_gt "24.0.0" $EMACSVER`  ]]; then
```

Comment:
See above comment at `version_gt` function definition.

Reply:
I'll fix it.

Reply:
See 3cbcdfe